agent/api/task/task_windows_test.go

```diff
@@ -254,6 +257,52 @@ func TestCPUPercentBasedOnUnboundedEnabled(t *testing.T) {
 	}
 }
 
+func TestWindowsMemoryReservationOption(t *testing.T) {
+	// Testing sending a task to windows overriding MemoryReservation value
+	rawHostConfigInput := dockercontainer.HostConfig{
+		Resources: dockercontainer.Resources{
+			MemoryReservation: nonZeroMemoryReservationValue,
+		},
+	}
+
+	rawHostConfig, err := json.Marshal(&rawHostConfigInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testTask := &Task{
+		Arn:     "arn:aws:ecs:us-east-1:012345678910:task/c09f0188-7f87-4b0f-bfc3-16296622b6fe",
+		Family:  "myFamily",
+		Version: "1",
+		Containers: []*apicontainer.Container{
+			{
+				Name: "c1",
+				DockerConfig: apicontainer.DockerConfig{
+					HostConfig: strptr(string(rawHostConfig)),
+				},
+			},
+		},
+		PlatformFields: PlatformFields{
+			RamUnbounded: false,
+		},
+	}
+
+	// With RamUnbounded set to false, MemoryReservation is not overridden
+	config, configErr := testTask.DockerHostConfig(testTask.Containers[0], dockerMap(testTask), defaultDockerClientAPIVersion)
+	if configErr != nil {
```

Comment:
this can be assert. same below

Reply:
you can use require.NoError(t, err) if you want the exit behavior you get from t.fatal

Reply:
https://github.com/stretchr/testify/issues/562 

😞